rhel/resolver.go

```diff
@@ -0,0 +1,56 @@
+package rhel
+
+import (
+	"context"
+
+	"github.com/quay/zlog"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/indexer"
+)
+
+var (
+	_ indexer.Resolver = (*Resolver)(nil)
+)
+
+type Resolver struct{}
+
+// Resolve takes a claircore.IndexReport and uses the rpm Files
+// to determine if the package originate from an RPM. If the package
+// was deemed to have been installed via RPM it isn't included in the
+// final report.
+func (r *Resolver) Resolve(ctx context.Context, ir *claircore.IndexReport, layers []*claircore.Layer) *claircore.IndexReport {
+	for pkgID, pkg := range ir.Packages {
+		isRPMPackage := false
+	envLoop:
+		for i := 0; i < len(ir.Environments[pkgID]); i++ {
+			if ir.Environments[pkgID][i].RepositoryIDs != nil {
+				for _, rID := range ir.Environments[pkgID][i].RepositoryIDs {
+					r := ir.Repositories[rID]
+					if r.Key == repositoryKey {
```

Comment:
Does this happen? I was under the impression if we currently see a JAR file, wouldn't the only repo be the maven one?

Reply:
This block is checking if the package is an RHEL RPM package, and negating any exclusion if it is. In reality this is a "belt and braces" check to make sure that no rpm packages are discounted from the final index report but I don't think we set a Filepath when we index these packages at the moment (but could in the future)

Reply:
Oh ok got it. I misremembered how the `filesFromDB` function worked